lms/djangoapps/commerce/utils.py

```diff
@@ -109,3 +117,264 @@ def upgrade_url(self, user, course_key):
+    if not is_commerce_service_configured():
+        log.error(
+            'Ecommerce service is not configured, cannot refund for user [%s], course entitlement [%s].',
+            enrollee.id,
+            entitlement_uuid
+        )
+        return False
+
+    service_user = user_model.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+    api_client = ecommerce_api_client(service_user)
+
+    log.info(
+        'Attempting to create a refund for user [%s], course entitlement [%s]...',
+        enrollee.id,
+        entitlement_uuid
+    )
+
+    try:
+        refund_ids = api_client.refunds.post(
+            {
+                'order_number': course_entitlement.order_number,
+                'username': enrollee.username,
+                'entitlement_uuid': entitlement_uuid,
+            }
+        )
+    except Exception as exc:  # pylint: disable=broad-except
+        # Catch any possible exceptions from the Ecommerce service to ensure we fail gracefully
+        log.exception(
+            "Unexpected exception while attempting to initiate refund for user [%s], "
+            "course entitlement [%s] message: [%s]",
+            enrollee.id,
+            course_entitlement.uuid,
+            str(exc)
+        )
+        return False
+
+    if refund_ids:
+        log.info(
+            'Refund successfully opened for user [%s], course entitlement [%s]: %r',
+            enrollee.id,
+            entitlement_uuid,
+            refund_ids,
+        )
+
+        return _process_refund(
+            refund_ids=refund_ids,
+            api_client=api_client,
+            mode=course_entitlement.mode,
+            user=enrollee,
+            always_notify=True,
+        )
+    else:
+        log.warn('No refund opened for user [%s], course entitlement [%s]', enrollee.id, entitlement_uuid)
+        return False
+
+
+def refund_seat(course_enrollment):
+    """
+    Attempt to initiate a refund for any orders associated with the seat being unenrolled,
+    using the commerce service.
+
+    Arguments:
+        course_enrollment (CourseEnrollment): a student enrollment
+
+    Returns:
+        A list of the external service's IDs for any refunds that were initiated
+            (may be empty).
+
+    Raises:
+        exceptions.SlumberBaseException: for any unhandled HTTP error during communication with the E-Commerce Service.
+        exceptions.Timeout: if the attempt to reach the commerce service timed out.
+    """
+    User = get_user_model()  # pylint:disable=invalid-name
+    course_key_str = unicode(course_enrollment.course_id)
+    enrollee = course_enrollment.user
+
+    service_user = User.objects.get(username=settings.ECOMMERCE_SERVICE_WORKER_USERNAME)
+    api_client = ecommerce_api_client(service_user)
+
+    log.info('Attempting to create a refund for user [%s], course [%s]...', enrollee.id, course_key_str)
+
+    refund_ids = api_client.refunds.post({'course_id': course_key_str, 'username': enrollee.username})
```

Comment:
Note: The bug fix and change made to fix the refund failure on stage

This was enrollee.id in my last PR